### ISSUE TITLE
resolvectl flush-caches seems not to work in a consistent manner

### ISSUE THREAD

**Issue by q2dg** [systemd/systemd#12243]
Systemd version where I've seen the explained behaviour is v239 (which is last version available in Fedora 29)

I have in my net lab a Bettercap instance (www.bettercap.org) doing an ARP-spoofing to be able to spoof DNS responses from a specific domain (say, "www.marca.com"). I can see with Wireshark the arp-spoofing is running ok. 

If in the "victim" computer I execute _dig @ip.Bettercap www.marca.com_, I get the spoofed IP, directly from "bad" Bettercap's DNS server, right. But when I execute _dig www.marca.com_  I get instead the correct non-spoofed answer from systemd-resolved (127.0.0.53) , so I infer this answer is in systemd-resolved cache. After doing _resolvectl flush-caches_  what I would expect is receiving the spoofed DNS answer when executing _dig www.marca.com_ again but, instead, I continue receiving the "good" answer. And my wireshark shows there haven't been any attempt to do any request, so it must be a cache affair.

What it's a lot more strange is that after many times repeteadly executing _resolvectl flush-caches_ , sometimes and magically, _dig www.marca.com_ resolves to spoofed "bad" IP, but happiness is short because a few seconds later the answer turns to be the real one again (???). I suspect that other non-spoofed DNS requests to other unrelated domains may bypass the current systemd-resolved`s cache content....it's very strange.

In summary, I think systemd-resolved cache behaviour deserves a look.
Thanks a lot

**Comment by poettering** [comment]
What's the "resolvectl status" output you see when this happens?

**Comment by q2dg** [comment]
Sorry, I'll close the issue because I can't re-reproduce the explained "bad" behaviour. I'll reopen it when I have more relevant information. Sorry again
